pnpm-sync-api-demo/src/index.ts

```diff
@@ -0,0 +1,3 @@
+export function demo(): void {
```

Comment:
It would make more sense to name this `start.ts` and have it be an executable program instead of an exported API

Reply:
Thinking about the bigger picture, though:

If someone is looking for a basic illustration of how to invoke your API, the `packages/pnpm-sync/src/start.ts` CLI already provides a fairly intuitive example.  It's missing the details of how to do a more advanced integration such as formatting messages with color, although you could address that by simply linking to the Rush source code that calls these APIs, which I think is fairly readable.

On the other hand, someone wants to see a concrete example of WHAT the tool does, your Jest tests may be a fairly good illustration since they read/write a realistic directory structure.  So a possible different direction might be to rename `pnpm-sync-api-demo` to `tests/pnpm-sync-api-test`.  And then your **README.md** docs could contain manual steps showing how to manually invoke the `pnpm-sync` CLI under the `test-fixtures` folder, and explain what files get written, and what it means.  Then it would be understandable that the Jest tests effectively automate just this manual tutorial and confirm its outputs.

Reply:
Thanks for detailed suggestion. 
Original intention was to utilize the `pnpm-sync-api-demo` to test the `pnpm-sync-lib` project. Maybe, it is a not good idea to combine them. 
So, I will refactor the folder structure to create a tests folder, and put all necessary tests there. And, if there is a need to create a demo project on how to use `pnpm-sync-lib` APIs,  we can add it later.